Initial front end with database integration

Imran2923 opened the pull request:
Updated code to use SQL server instead of internal server and shows components based on user logged in.

mysqlclient has to now be installed on the client machine to be able to connect to the database through front end and server has to be running for the website to fully work.

ariscahyadi commented on web/nclsite/settings.py:
```diff
@@ -77,8 +76,12 @@
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.mysql',
```
@Imran2923 can you check with @jannemartini ... because Martini gave me another value as mentioned here https://github.com/nus-ncl/CTFPro/pull/16/files#diff-4064389f8f6a8fa597a4b180604d57d99a920a9c04be2e7cdda25fcb1ca4d846R71

ariscahyadi reviewed (commented):
I have small things to check.

Imran2923 commented:
Updated more code to include form to add components to database. More improvements and bug fixes too. Changed models so that database will have the correct tables once makemigrations command is made.
